### ISSUE TITLE
Token error in minified shaders

### ISSUE THREAD

**Issue by UstymUkhman** [brendan-duncan/wgsl_reflect#56]
Hi! First of all, this is a great tool, it makes extracting vertex attributes and shader uniforms so much easier. Thank you for that!

I've run into an issue I'm not sure if possible to solve, so I'd like your opinion on this one.
I have a tool to minify GLSL and WGSL shaders, but when used in combination with `wgsl_reflect`, it may cause one of the errors shown in the screenshots. The cause seems to be the same: `-1`.

For example, this is a valid, minified shader produced by [`vite-plugin-glsl`](https://github.com/UstymUkhman/vite-plugin-glsl):

```wgsl
@group(0)@binding(0)var<uniform>resolution: vec2f;fn GetClipSpace(position: vec2f)->vec2f{let clipSpace=position/resolution*2-1;return clipSpace*vec2f(1,-1);}
```

but the error is caused for `2-1` being attached together. The only thing required to make it work was to separate them with a space:

```wgsl
@group(0)@binding(0)var<uniform>resolution: vec2f;fn GetClipSpace(position: vec2f)->vec2f{let clipSpace=position/resolution*2 - 1;return clipSpace*vec2f(1,-1);}
```

I have several other shaders and the fix was always the same. Strangely enough, other operations (`*`,`/` and `+`) cause no errors when two operands are attached together. Is it possible to do something with this or is it some limitation of the parsing algorithm? As a workaround for me, it's possible to disable minification all together, but of course, I'd prefere not to. 😄 

Thanks!

![Screenshot 2024-08-28 232101](https://github.com/user-attachments/assets/a7c7489a-eb29-413d-8dbf-b636ef8729e1)

![Screenshot 2024-08-28 231328](https://github.com/user-attachments/assets/12425a2b-a3e2-4ec5-b2ea-8f3b4a1e976a)

**Comment by brendan-duncan** [comment]
Thanks for the report, I'll get it fixed up as soon as I get a chance. Shouldn't be too hard to fix.

**Comment by brendan-duncan** [comment]
I pushed a fix to git, but I'll have to wait until later to do more testing and publish, have to get back to my real job for a bit.
The lexer tries to disambiguate tokens that can be interpreted different ways depending on context, and it wasn't handling "-" very well. 1-2 was parsed as "1", "-2", instead of the correct "1", "-", "2". Hopefully it does a better job now.

**Comment by brendan-duncan** [comment]
Updated the published version with the fixes as 1.0.10.

**Comment by UstymUkhman** [comment]
Thank you so much for such a quick fix. It works like a charm! And again: great library, thanks for working on it!
